### PR TITLE
feat: Add support for .NET agent lambdas using the layer.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Adds support for .NET lambda layers, mainly by setting up the environment variables.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ const wrappableRuntimeList = [
   "java11",
   "java17",
   "java21",
+  "dotnet6",
+  "dotnet7",
```
Comment (chaudharysaket) [inline src/index.ts:35]: Is `dotnet7` supported?

File: src/index.ts [hunk 2]
```diff
@@ -537,6 +540,14 @@ or make sure that you already have Serverless 3.x installed in your project.
       environment.NEW_RELIC_SERVERLESS_MODE_ENABLED = "true";
     }
 
+    // Uses same layer as CLI so the paths will be the same
```
Comment (chaudharysaket) [inline src/index.ts:543]: Could you please look into the `lint` issue because of which the checks are failing?

**Review by chaudharysaket — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (nr-ahemsath) [reply src/index.ts:35]: It depends on your definition of "supported":
1. .NET 7 is no longer supported by Microsoft as of 5/14/24, so ideally customers shouldn't be using it.  However...
2. There's no technical reason why our Lambda instrumentation won't work for a .NET 7 Lambda function.

So, we think it's best to be as inclusive as possible here.

File: src/index.ts (hunk 2 above)
Comment (nr-ahemsath) [reply src/index.ts:543]: Working on it.

**Comment by ashishsinghnr** [comment]
@nr-ahemsath , 
1. for validate pull request. Please refer to the following document - https://www.conventionalcommits.org/en/v1.0.0/
Example - <type>[optional scope]: <description>
2.  Also, kindly re-validate the test cases, looks like  test cases are failing.

**Review by mrickard — APPROVED** [review]

**Pull request merged**